.circleci/config.yml

```diff
@@ -71,13 +71,54 @@ jobs:
             - "./target"
             - "./pallets/runtime/target"
             - "./pallets/runtime/wasm/target"
+  cli:
+    docker:
+      - image: maxsam4/rust
+    environment:
+      VERBOSE: "1"
+    steps:
+      - checkout
+      - run:
+          name: Build concatenated lock file for caching
+          command: find . -name "Cargo.lock" -not -path "*/target/*" | xargs cat > concatenated.lock
+      - run:
+          name: Store rust version in a file for cache key
+          command: rustc --version > rust.version
+      - restore_cache:
+          keys:
+            - debug-cache-{{ checksum "./rust.version" }}-{{ checksum "./pallets/runtime/Cargo.lock" }}-{{ checksum "./concatenated.lock" }}
+            - debug-cache-{{ checksum "./rust.version" }}-{{ checksum "./pallets/runtime/Cargo.lock" }}
+            - debug-cache-{{ checksum "./rust.version" }}
+            - debug-cache
```

Comment:
please use `release` instead of `debug` in all the above cache keys (and below in save cache) because you are building the node in release mode not debug mode.